frontend/src/utils/CallbackUtils.ts

```diff
@@ -0,0 +1,9 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const deferCallback = <T extends (...args: any[]) => any>(f: T) => (
+  ...args: Parameters<T>
+): ReturnType<T> => f(...args);
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const makeDeferredCallback = <T extends (...args: any[]) => any>(
+  f: T,
+) => (...args: Parameters<T>) => (): ReturnType<T> => f(...args);
```

Comment:
Maybe it'd be good to write a comment about how this works or where we got it from? I see it's used in `CourseCard` to set the hover and focused states. What does it do?

Reply:
const focus = () => setFoo(3);